hypothesis-python/src/hypothesis/control.py

```diff
@@ -125,3 +128,61 @@ def event(value):
 
     if context.data is not None:
         context.data.note_event(value)
+
+
+def target(observation, label=""):
+    # type: (float, str) -> None
+    """Calling this function with a ``float`` observation gives it a way
+    to do a directed, rather than heuristic or random, search for inputs
```

Comment:
Nit: I'm not wild about this phrasing, as our search is both random and heuristic even when using `target`.  Maybe something more neutral like "guides test case generation towards examples which result in a higher target score"?